packages/react-link/src/components/Link/useLinkState.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { LinkState } from './Link.types';
+
+/**
+ * The useLink hook processes the Link draft state.
+ * @param draftState - Link draft state to mutate.
+ */
+export const useLinkState = (draftState: LinkState) => {
+  const { as, disabled, disabledFocusable, href, onClick, type } = draftState;
+
+  draftState.onClick = (ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement>) => {
```

Comment:
I think a space keydown handler is necessary to emulate native link

Reply:
I'll play with this and circle back as I'm not sure if it is entirely necessary.

Reply:
natively `a` and `button` will be fine, the issue is that we support `as` :/

Reply:
Ahh, yeah, makes sense, I don't know how I forgot that, thanks!